compilers/concrete-compiler/compiler/lib/Support/CompilationFeedback.cpp

```diff
@@ -6,60 +6,115 @@
 #include <cassert>
 #include <fstream>
 
-#include "llvm/Support/JSON.h"
-
 #include "concretelang/Support/CompilationFeedback.h"
 
+using concretelang::protocol::Message;
+
 namespace mlir {
 namespace concretelang {
 
-void CompilationFeedback::fillFromClientParameters(
-    ::concretelang::clientlib::ClientParameters params) {
+inline size_t bitWidthAsWord(size_t exactBitWidth) {
+  if (exactBitWidth <= 8)
+    return 8;
+  if (exactBitWidth <= 16)
+    return 16;
+  if (exactBitWidth <= 32)
+    return 32;
+  if (exactBitWidth <= 64)
+    return 64;
+  assert(false && "Bit witdh > 64 not supported");
+}
+
+void CompilationFeedback::fillFromProgramInfo(
+    const Message<concreteprotocol::ProgramInfo> &programInfo) {
+  auto params = programInfo.asReader();
+
   // Compute the size of secret keys
   totalSecretKeysSize = 0;
-  for (auto sk : params.secretKeys) {
-    totalSecretKeysSize += sk.byteSize();
+  for (auto skInfo : params.getKeyset().getLweSecretKeys()) {
+    assert(skInfo.getParams().getIntegerPrecision() % 8 == 0);
```

Comment:
Is this really supposed to be a check for a multiple of 8 or should this be a check for a power of two?

Reply:
Well to compute `totalSecretKeysSize` correctly, checking that it is a multiple of 8 is what we want yes.